CONTRIBUTING.md

```diff
@@ -0,0 +1,202 @@
+
+## Guidelines for your contributions
+
+To ensure that making a contribution is a positive experience for both contributor and reviewer we ask that you read and follow these community guidelines.
+
+This communicates that you respect our time as developers. We will return
+that respect by addressing your issues, assessing proposed changes and
+finalising your pull requests, as helpfully and efficiently as possible.
+
+These are mostly guidelines, not rules. Use your best judgement and feel free to
+propose changes to this document in a pull request.
+
+<!-- Include end contributing intro -->
+
+## Quicklinks
+
+* [Code of conduct](#code-of-conduct)
+* [Getting started](#getting-started)
+* [Issues](#issues)
+* [Pull requests](#pull-requests)
+* [Contributing to the code](#contributing-to-the-code)
+* [Contributing to the docs](#contributing-to-the-docs)
+* [Contributor License Agreement](#contributor-license-agreement)
+* [Getting kelp](#getting-help)
+
+<!-- Include start contributing main -->
+
+## Code of conduct
+
+We take our community seriously and hold ourselves and other contributors to high standards of communication. By participating and contributing, you agree to uphold the Ubuntu Community [Code of Conduct](https://ubuntu.com/community/code-of-conduct).
+
+## Getting started
+
+The source code for Ubuntu WSL and Ubuntu Pro for WSL can be found on GitHub:
+
+- [Ubuntu WSL repo](https://github.com/ubuntu/WSL) 
+- [Ubuntu Pro for WSL repo](https://github.com/canonical/ubuntu-pro-for-wsl) 
+
+### Issues with WSL should be directed to Microsoft's WSL project
+
+<!-- TODO: a breakdown of responsibilities or example issues (Ubuntu/Microsft) could be helpful here -->
+
+We accept any contributions relating to Ubuntu WSL and Ubuntu Pro for WSL.
+However, we do not directly maintain WSL itself, which is a Microsoft product.
+If you have identified a problem or bug in WSL then file an issue in
+[Microsoft's WSL project repository](https://github.com/microsoft/WSL/issues/).
+
+If you are unsure whether your problem relates to an Ubuntu project or the Microsoft project then familiarise yourself with their documentation.
+
+- [Ubuntu WSL docs](https://canonical-ubuntu-wsl.readthedocs-hosted.com/en/latest/)
+- [Microsoft WSL docs](https://learn.microsoft.com/en-us/windows/wsl/)
+
+At this point, if you are still not sure, try to contact a maintainer of one of the projects who will advise you where best to submit your Issue.
+
+### How to contribute
+
+Contributions are made via Issues and Pull Requests (PRs). A few general guidelines that cover both:
+
+* Use the advisories page of the repository and not a public bug report to
+report security vulnerabilities. For Ubuntu WSL, please use [launchpad private
+bugs](https://bugs.launchpad.net/ubuntuwsl/+filebug) which is
+monitored by our security team. On an Ubuntu machine, it’s best to use
+`ubuntu-bug wsl` to collect relevant information.
+<!-- QUERY: is `wsl` the correct argument here???. -->
```

Comment:
Actually we cannot recommend this. It would:
1. require that the user is using an Ubuntu machine as you mention, which we cannot expect since they are likely running Windows.
2. report a bug to wsl in Launchpad which is Wsman Shell command Line. So totally unrelated to our WSL (WSL is also the World Surf League but nothing to do with our topic ;) )

I would recommend to report issues directly to the GitHub project.

Reply:
That's fine, I can recommend reporting issues directly on GitHub.

I thought the original wording might need to be removed. Initially, I had `ubuntu-bug ubuntuwsl`, following the launchpad link above but that did not seem work. I wasn't aware of Wsman Shell.

Also, I thought in this case "ubuntu machine" would include ubuntu running on WSL, because the `ubuntu-bug` command can be run in Ubuntu WSL.